types/types.go

```diff
@@ -56,12 +56,15 @@ type Image interface {
 // ImageManifest is the interesting subset of metadata about an Image.
 // TODO(runcom)
 type ImageManifest interface {
+	Raw() []byte
+	Layers() []string
 	String() string
 }
 
 // DockerImageManifest is a set of metadata describing Docker images and their manifest.json files.
 // Note that this is not exactly manifest.json, e.g. some fields have been added.
 type DockerImageManifest struct {
```

Comment:
It seems to me that `inspect` should define its own struct (with whatever fields it wants, coming from whatever mutlitude of sources needed) separate from this library interface, then.

Having `Labels`, which come from the config stored inside the manifest, is very useful and completely fine; having `RepoTags`, which have very little to do with the manifest, is very strange, and even the `Name` and `Tag` (no longer present in v2s2) have unclear properties. When an image which calls itself `library/busybox:latest` is being pulled from `private.mitr/l33t/busybox:$timestamp and pushed to `library/busybox:theNextGeneration`, does it make sense to talk about “the” name and tag? (OTOH it still makes perfect sense to talk about “the” labels and layers.)